thrust/testing/unittest/random.h

```diff
@@ -92,8 +69,8 @@ THRUST_NS_QUALIFIER::host_vector<T> random_integers(const size_t N)
 {
   THRUST_NS_QUALIFIER::host_vector<T> vec(N);
   THRUST_NS_QUALIFIER::transform(
-    THRUST_NS_QUALIFIER::counting_iterator<unsigned int>(static_cast<unsigned int>(0)),
-    THRUST_NS_QUALIFIER::counting_iterator<unsigned int>(static_cast<unsigned int>(N)),
+    THRUST_NS_QUALIFIER::counting_iterator(static_cast<unsigned int>(0)),
```

Comment:
Style: I often prefer list-initialization to disambiguate function calls
```suggestion
    THRUST_NS_QUALIFIER::counting_iterator{static_cast<unsigned int>(0)},
```